### DISCUSSION TITLE
PSSecurityException when running Import-Module Pester

### DISCUSSION THREAD

**Discussion by santhoshmprabhu** [pester/Pester#2595] [Q&A] [answered]
Hi, I am new to Pester, and also dev on Windows in general. I am trying to use Pester to run tests on a different repo that I am working on, I seem to get the following error each time:

```
Import-Module : File C:\Program Files\WindowsPowerShell\Modules\Pester\5.6.1\Pester.psm1 cannot be loaded. The contents of
file C:\Program Files\WindowsPowerShell\Modules\Pester\5.6.1\Pester.psm1 might have been changed by an unauthorized user or
process, because the hash of the file does not match the hash stored in the digital signature. The script cannot run on the
specified system. For more information, run Get-Help about_Signing..
At line:1 char:1
+ Import-Module Pester
+ ~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : SecurityError: (:) [Import-Module], PSSecurityException
    + FullyQualifiedErrorId : UnauthorizedAccess,Microsoft.PowerShell.Commands.ImportModuleCommand
```

I am using Windows Powershell. I have tried installing Pester from the PSGallery as well as from chocolatey, I get the same error. I also tried 5.6.0 from chocolatey to no avail.  Could someone kindly provide any pointers? More info about my Shell:

```
Name                           Value
----                           -----
PSVersion                      5.1.26100.2161
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.26100.2161
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

EDIT: I subsequently tried Powershell, and can now invoke Pester. That does unblock me, but still can't get Pester to work on Windows Powershell.

**Comment by fflaten** [comment]
That's weird. Could you verify that it's loading the same module in both Windows PowerShell and PowerShell? One might have been modified. Does `Import-Module Pester; Get-Module Pester | Format-List Name, Version, Path` return the same path in both?

**Reply by santhoshmprabhu** [reply]
I haven't manually edited anything, not sure if there's a setting somewhere in my machine that's causing this.

Here's the output on Powershell:
```
PS C:\Users\sanprabhu> Import-Module Pester; Get-Module Pester | Format-List Name, Version, Path

Name    : Pester
Version : 5.6.1
Path    : C:\Program Files\WindowsPowerShell\Modules\Pester\5.6.1\Pester.psm1
```
On windows powershell, I run into the original error of course, but the error does begin with:
```
Import-Module : File C:\Program Files\WindowsPowerShell\Modules\Pester\5.6.1\Pester.psm1 cannot be loaded.
```
So the path seems to be the same.

**Reply by fflaten** [reply]
Weird. Have you tried reinstalling the module?

**Reply by santhoshmprabhu** [reply]
I have tried installing from PSGallery a few times, and as I mentioned, tried two versions from chocolatey.

**Reply by fflaten** [reply]
Sorry, I missed that reading on mobile. Which locale are you using in Windows? You can use `Get-WinSystemLocale`. Which ExecutionPolicy are you using in Windows PowerShell?

**Reply by santhoshmprabhu** [reply]
Windows Powershell
```
PS C:\Users\sanprabhu> Get-WinSystemLocale

LCID             Name             DisplayName
----             ----             -----------
1033             en-US            English (United States)


PS C:\Users\sanprabhu> Get-ExecutionPolicy
AllSigned
```

Powershell
```
PS C:\Users\sanprabhu> Get-WinSystemLocale

LCID             Name             DisplayName
----             ----             -----------
1033             en-US            English (United States)

PS C:\Users\sanprabhu> Get-ExecutionPolicy
RemoteSigned
```

I'll look into updating the execution policy.

**Reply by fflaten** [reply]
Thanks for catching this. I'm able to reproduce it now and there's a hash mismatch. Weird that it only affects Windows PowerShell when PowerShell also detects the mismatch. Moving to an issue.

**Comment by nohwnd** [comment, marked as answer]
The files in 5.6.x have a broken signature, that is why `AllSigned` execution policy refuses to load them. We could not fix the already published versions, 5.7.0 and newer are signed correctly, so updating Pester fixes it.

Tracked in #2596.
